sdk/typespec/typespec_client_core/examples/binary_data_request.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use tokio::fs;
+use tracing::level_filters::LevelFilter;
+use tracing_subscriber::fmt::format::FmtSpan;
+use typespec_client_core::{fs::FileStreamBuilder, http::RequestContent};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Log traces to stdout.
+    let _log = tracing_subscriber::fmt()
+        .with_span_events(FmtSpan::ENTER | FmtSpan::EXIT)
+        .with_max_level(LevelFilter::DEBUG)
+        .init();
+
+    // Asynchronously read the whole file into memory.
+    let body = RequestContent::from(fs::read(file!()).await?);
+    client::put_binary_data(body).await?;
+
+    // Asynchronously stream the file with the service client request.
+    #[cfg(not(target_arch = "wasm32"))]
+    {
+        let file = fs::File::open(file!()).await?;
+        let file = FileStreamBuilder::new(file)
+            // Simulate a slow, chunky request.
+            .buffer_size(256usize)
+            .build()
+            .await?;
+        client::put_binary_data(file.into()).await?;
+    }
+
+    Ok(())
+}
+
+mod client {
+    use futures::StreamExt;
+    use tracing::debug;
+    use typespec_client_core::{
+        http::{headers::Headers, Body, RequestContent, Response, StatusCode},
+        stream::BytesStream,
+    };
+
+    #[tracing::instrument(skip(body))]
+    pub async fn put_binary_data(
+        body: RequestContent<()>,
+    ) -> typespec_client_core::Result<Response<()>> {
+        let body: RequestContent<()> = body.into();
+        let body: Body = body.into();
+
+        let content = match body {
+            Body::Bytes(ref bytes) => {
+                debug!("received bytes");
+                bytes.to_owned()
+            }
+            Body::SeekableStream(mut stream) => {
+                debug!("received stream");
+                let stream = stream.as_mut();
+
+                let mut bytes = Vec::new();
+                while let Some(buf) = stream.next().await {
+                    debug!("reading partial request...");
+                    bytes.extend(buf?);
+                }
+
+                bytes.into()
+            }
+        };
+
+        // Assume bytes are a string in this example.
+        let content = String::from_utf8(content.into())?;
+        println!("{content}");
```

Comment:
Any risk here of dumping PII? Should there be a note surrounding this println about contents could be senstive in nature?

Reply:
There's no service client here. It's literally this source file. But, if you're comment is more about warning for real-world use cases I could.